### PR TITLE
Do not send Log-list updates when no update passed.

### PR DESCRIPTION
**Pull request by Mercurrent**

### CODE REVIEW

File: submission/loglist_manager_test.go [hunk 1]
```diff
@@ -67,3 +69,49 @@ func TestFirstRefresh(t *testing.T) {
...
+
+	sampleLogListUpdate := strings.Replace(testdata.SampleLogList, "ct.googleapis.com/racketeer/", "ct.googleapis.com/racketeer/v2/", 1)
+	if err := ioutil.WriteFile(f, []byte(sampleLogListUpdate), 0644); err != nil {
+		t.Fatalf("unable to update Log-list data file: %q", err)
+	}
+	select {
+	case <-llm.LLUpdates:
+		return
+	case err := <-llm.Errors:
+		t.Errorf("llm.Run() returned error %q while expected none", err)
```
Comment (RJPercival) [inline submission/loglist_manager_test.go]:
```suggestion
		t.Errorf("llm.Run() emitted error %q while expected none", err)
```

File: submission/loglist_manager_test.go [hunk 2]
```diff
@@ -67,3 +69,49 @@ func TestFirstRefresh(t *testing.T) {
...
+				readFirst = true
+			} else {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
+			}
+		case err := <-llm.Errors:
+			t.Errorf("llm.Run() returned error %q while expected none", err)
+		case <-halfCtx.Done():
+			break First
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply submission/loglist_manager_test.go]: Added the check for readFirst

File: submission/loglist_manager_test.go [hunk 3]
```diff
@@ -67,3 +69,49 @@ func TestFirstRefresh(t *testing.T) {
...
+	defer cancel()
+	halfCtx, halfCancel := context.WithTimeout(context.Background(), time.Second)
+	defer halfCancel()
+
+	llm.Run(ctx, 300*time.Millisecond)
+	readFirst := false
+First:
+	for {
+		select {
+		case <-llm.LLUpdates:
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply submission/loglist_manager_test.go]: Here it cannot be nil: the channel sends structs, not pointers to them.

File: submission/loglist_manager_test.go [hunk 4]
```diff
@@ -67,3 +69,49 @@ func TestFirstRefresh(t *testing.T) {
...
+			if !readFirst {
+				readFirst = true
+			} else {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
+			}
+		case err := <-llm.Errors:
+			t.Errorf("llm.Run() returned error %q while expected none", err)
```
[the remark this answers is not in the document]
Comment (Mercurrent) [reply submission/loglist_manager_test.go]: Done

File: submission/loglist_manager_test.go (hunk 1 above)
Comment (Mercurrent) [reply submission/loglist_manager_test.go]: Done

**Comment by Mercurrent** [comment]
PTAL

File: submission/loglist_manager_test.go [hunk 5]
```diff
@@ -62,7 +64,56 @@ func TestFirstRefresh(t *testing.T) {
...
+			if !readFirst {
+				readFirst = true
+			} else {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
+			}
+		case err := <-llm.Errors:
+			t.Errorf("llm.Run() remitted error %q while expected none", err)
+		case <-halfCtx.Done():
+			if !readFirst {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
```
Comment (RJPercival) [inline submission/loglist_manager_test.go]: This message looks like a copy-paste error. In this case, if `readFirst == false`, it hasn't emitted any updates at all.

Comment (Mercurrent) [reply submission/loglist_manager_test.go]: Thanks for the catch. Updated the message.

**Pull request merged**